Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py

```diff
@@ -246,6 +246,12 @@ def make_slice_ws(self, slice_list, x_data, x_unit, name):
             VerticalAxisValues=axis_names,
         )
 
+    def set_label(self, ws_str, label, unit):
+        ws = ADS.retrieve(ws_str)
+        axis = ws.getAxis(0)
+        axis.setUnit("Label").setLabel(label, unit)
+        return ws_str
```

Comment:
Why does this function need to return `ws_str`?

Reply:
so we can add it to the ws group on line 277

Reply:
should it be returning `ws` instead of `ws_str`?

Reply:
it works with a string and it makes the mocking nicer